code/src/components/Main.js

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from "react";
+import ThoughtForm from "./ThoughtForm";
+import ThoughtCard from "./ThoughtCard";
+import Spinner from "./Spinner";
+import { API_URL, LIKES_URL } from "../utils/urls";
+
+const Main = () => {
+  /* useStates */
+  const [thoughts, setThoughts] = useState([]); //It b
+  const [newThought, setNewThought] = useState(""); // It being used in the form to send the new message
+  const [loading, setLoading] = useState(false);
+
+  /* useEffects: it renders after the component gets mounted */
+  useEffect(() => {
+    fetchThoughts();
+  }, []);
+
+  const fetchThoughts = () => {
+    setLoading(true);
+    fetch(API_URL)
+      .then((res) => res.json())
+      .then((data) => setThoughts(data))
+      .finally(() => setLoading(false));
```

Comment:
I haven't worked with .finally so far. What are the advantages of using finally instead of setting the loading to false in the 2nd .then?

Reply:
Good question Reinis!  As far as I understand,  the methods then and finally are almost the same with the exception that when there is an error, the method finally will always execute but "then" will not. In this particular case, I am not sure either the advantage of setting the SetLoading in the finally function, I followed Maxsymilian's example but I am so intrigued now so I will ask this question in the FQ session on Friday so hopefully, we get the answer.